cmd/kiam-check/lambda.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	log "github.com/sirupsen/logrus"
+)
+
+// createLambdaClient creates and returns an AWS Lambda client
+func createLambdaClient(s *session.Session, region string) *lambda.Lambda {
+
+	// build a lambda client
+	log.Infoln("Building Lambda client for " + region + " region.")
+	return lambda.New(s, &aws.Config{Region: aws.String(region)})
+}
+
+// listLambdas returns a list of Lambda functions.
+func listLambdas(c *lambda.Lambda) ([]*lambda.FunctionConfiguration, error) {
+
+	// Create an output object for the list query.
+	list := make([]*lambda.FunctionConfiguration, 0)
+
+	// Query Lambdas.
+	log.Infoln("Querying Lambda functions.")
+
+	results, err := c.ListFunctions(&lambda.ListFunctionsInput{
+		MaxItems: aws.Int64(100),
+	})
+	if err != nil {
+		return list, err
+	}
+	log.Infoln("Queried", len(results.Functions), "Lambdas.")
+	list = append(list, results.Functions...)
+
+	// Keep querying until the API gives us everything.
+	var marker string
+	for results.NextMarker != nil {
+		log.Debugln("There are more results to be queried.")
+
+		marker = *results.NextMarker
+		results, err = c.ListFunctions(&lambda.ListFunctionsInput{
+			MaxItems: aws.Int64(100),
+			Marker:   aws.String(marker),
+		})
+		if err != nil {
+			return list, err
+		}
+
+		log.Infoln("Queried", len(results.Functions), "Lambdas.")
+		list = append(list, results.Functions...)
+	}
```

Comment:
hmm this marker thing seems to be be similar with querying for s3 objects as well... 🤔